internal/config/analytics.go

```diff
@@ -22,22 +24,25 @@ type AnalyticsStorageConfig struct {
 
 // ClickhouseConfig defines the connection details for connecting Flipt to Clickhouse.
 type ClickhouseConfig struct {
-	Enabled  bool   `json:"enabled,omitempty" mapstructure:"enabled" yaml:"enabled,omitempty"`
-	URL      string `json:"url,omitempty" mapstructure:"url" yaml:"url,omitempty"`
-	Username string `json:"-" mapstructure:"username" yaml:"username,omitempty"`
-	Password string `json:"-" mapstructure:"password" yaml:"password,omitempty"`
+	Enabled bool   `json:"enabled,omitempty" mapstructure:"enabled" yaml:"enabled,omitempty"`
+	URL     string `json:"-" mapstructure:"url" yaml:"url,omitempty"`
 }
 
 func (a *AnalyticsConfig) Enabled() bool {
 	return a.Storage.Clickhouse.Enabled
 }
 
-func (c *ClickhouseConfig) Auth() clickhouse.Auth {
-	return clickhouse.Auth{
-		Username: c.Username,
-		Password: c.Password,
-		Database: "flipt_analytics",
+// Options returns the connection option details for Clickhouse.
+func (c *ClickhouseConfig) Options() (*clickhouse.Options, error) {
+	options, err := clickhouse.ParseDSN(c.URL)
+	if err != nil {
+		return nil, err
+	}
+	if options.Auth.Database != "flipt_analytics" {
```

Comment:
i think we want to allow them to use whatever db name they choose

Reply:
@markphelps We could do that. They would have to create that DB beforehand. I guess we can clearly state that in our docs

Reply:
can we try and parse the db name from the connection string?

if its there, then we use that, if not then we use this default?

Reply:
Yeah I think we definitely want them to create it first themselves. Like we do with the relational DB.